### PR TITLE
Fix seed

### PR DESCRIPTION
**Pull request by jjanvier**

### CODE REVIEW

File: Command/AbstractGenerateCommand.php [hunk 1]
```diff
@@ -59,4 +60,17 @@ protected function getTotalCount(array $config)
...
+    /**
+     * Create the output dir where fixtures will be generated if necessary.
+     *
+     * @param string $outputDir
+     */
+    protected function createOutputDirIfNecessary($outputDir)
+    {
+        $fs = new Filesystem();
+        if (!$fs->exists($outputDir)) {
+            $fs->mkdir($outputDir);
```
Comment (BitOne) [inline Command/AbstractGenerateCommand.php]: Hum, bad practice !
Your software shouldn't do more than it's expected to do.

And creating directory is not its responsability. Look at Unix command: apart mkdir or command with directory modificator (cp for exemple), no command will silently create a directory without user knowledge: they will fail if the directory does not exists. Also, it's the best way for the user to do not understand why the directory he has created is not populated as he made a typo in the config file and the command didn't tell him !

File: Command/GenerateFixtureCommand.php [hunk 2]
```diff
@@ -45,6 +45,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $totalCount = $this->getTotalCount($config);
 
         $outputDir = $config['output_dir'];
+        $this->createOutputDirIfNecessary($outputDir);
```
Comment (BitOne) [inline Command/GenerateFixtureCommand.php]: :fire:

**Comment by nidup** [comment]
gtm

File: Command/AbstractGenerateCommand.php (hunk 1 above)
Comment (jjanvier) [inline Command/AbstractGenerateCommand.php]: ok I remove it
but when you launch the command for the first time, it fails... that's a strange sensation

maybe the doc is bad

File: Generator/AssetCategoryGenerator.php [hunk 3]
```diff
@@ -29,11 +29,14 @@ class AssetCategoryGenerator implements GeneratorInterface
      *
      * {@inheritdoc}
      */
-    public function generate(array $config, $outputDir, ProgressHelper $progress, array $options = null)
+    public function generate(array $globalConfig, array $config, ProgressHelper $progress, array $options = [])
     {
         $this->locales = $options['locales'];
 
-        $faker = Faker\Factory::create();
+        $faker = \Faker\Factory::create();
```
Comment (BitOne) [inline Generator/AssetCategoryGenerator.php:36]: Shouldn't we provide a factory shared between generators instead of reusing the same code ?

**Pull request merged**